core/src/main/java/org/apache/gravitino/Entity.java

```diff
@@ -105,4 +106,147 @@ default void validate() throws IllegalArgumentException {
    * @return The type of the entity as defined by {@link EntityType}.
    */
   EntityType type();
+
+  /**
+   * Represents a relational entity that contains an entity, its vertex type, related identifiers,
+   * and the type of related entity.
+   *
+   * @param <E> the type of the entity, which must extend {@link Entity} and implement {@link
+   *     HasIdentifier}
+   */
+  interface RelationalEntity<E extends Entity & HasIdentifier> {
```

Comment:
You don't have to implement this `RelationalEntity`  under `Entity`, use a class file to define this.